stock_move_manual_lot/tests/test_stock_move_manual_lot.py

```diff
@@ -84,9 +84,32 @@ def test_01_force_manual_selection(self):
     def test_02_reassign_reservation(self):
         """Pickings are rereserved after their lots were fetched on another"""
         self.assertEqual(self.quant1.reserved_quantity, 1)
+        lot1 = self.picking.move_line_ids.lot_id
+        self.assertTrue(lot1)
         picking2 = self.picking.copy()
         self.assertFalse(self.quant2.reserved_quantity, 1)
         picking2.action_assign()
+        lot2 = picking2.move_line_ids.lot_id
+        self.assertTrue(lot2)
+        self.assertEqual(self.picking.state, 'assigned')
+        self.assertEqual(picking2.state, 'assigned')
+        self.assertEqual(self.picking.move_line_ids.lot_id, lot1)
+        self.assertEqual(picking2.move_line_ids.lot_id, lot2)
+
+        # set the lot from the other picking
+        picking2.write({
+            'move_line_ids_without_package': [[1, picking2.move_line_ids.id, {
+                'manual_lot_id': lot1.id
+            }]]})
+
+        # both pickings are still assigned
+        self.assertEqual(self.picking.state, 'assigned')
+        self.assertEqual(picking2.state, 'assigned')
+
+        # the lots are swapped
+        self.assertEqual(self.picking.move_line_ids.lot_id, lot2)
+        self.assertEqual(picking2.move_line_ids.lot_id, lot1)
+
         self.assertEqual(self.quant2.reserved_quantity, 1)
         self.assertEqual(picking2.move_line_ids.lot_id, self.lot2)
```

Comment:
Not sure how `self.lot2` relates to `lot2`, but tests fail on 
```
2022-01-07 09:33:58,814 204298 ERROR 12stock_move_manual_lot odoo.addons.stock_move_manual_lot.tests.test_stock_move_manual_lot: FAIL: test_02_reassign_reservation (odoo.addons.stock_move_manual_lot.tests.test_stock_move_manual_lot.TestStockMoveManualLot) 
2022-01-07 09:33:58,814 204298 ERROR 12stock_move_manual_lot odoo.addons.stock_move_manual_lot.tests.test_stock_move_manual_lot: ` Pickings are rereserved after their lots were fetched on another 
2022-01-07 09:33:58,814 204298 ERROR 12stock_move_manual_lot odoo.addons.stock_move_manual_lot.tests.test_stock_move_manual_lot: Traceback (most recent call last): 
2022-01-07 09:33:58,814 204298 ERROR 12stock_move_manual_lot odoo.addons.stock_move_manual_lot.tests.test_stock_move_manual_lot: `   File "/home/bruno/buildout/ou120/parts/stock-logistics-workflow/stock_move_manual_lot/tests/test_stock_move_manual_lot.py", line 114, in test_02_reassign_reservation 
2022-01-07 09:33:58,814 204298 ERROR 12stock_move_manual_lot odoo.addons.stock_move_manual_lot.tests.test_stock_move_manual_lot: `     self.assertEqual(picking2.move_line_ids.lot_id, self.lot2) 
2022-01-07 09:33:58,814 204298 ERROR 12stock_move_manual_lot odoo.addons.stock_move_manual_lot.tests.test_stock_move_manual_lot: ` AssertionError: stock.production.lot(1122,) != stock.production.lot(1123,) 
```

And I think that what Holger means is, the tests won't reproduce the problem unless the write date of the stock moves are backdated somewhere.